api/api_full.go

```diff
@@ -171,6 +171,10 @@ type FullNode interface {
 	// If oldmsgskip is set, messages from before the requested roots are also not included.
 	ChainExport(ctx context.Context, nroots abi.ChainEpoch, oldmsgskip bool, tsk types.TipSetKey) (<-chan []byte, error) //perm:read
 
+	ChainExportRange(ctx context.Context, head, tail types.TipSetKey, cfg *ChainExportConfig) (<-chan []byte, error) //perm:read
```

Comment:
https://github.com/filecoin-project/go-jsonrpc/pull/88 adds support for calling methods on the client.

A callback accepting bytes on the client would be better as that lets you have backpressure, and is arguably more robust in general.

Reply:
(see https://github.com/filecoin-project/lotus/pull/10027/files for an example of how it's integrated in lotus)